wallet/internal/db/interface.go

```diff
@@ -221,3 +222,31 @@ type UTXOStore interface {
 	Balance(ctx context.Context, params BalanceParams) (
 		btcutil.Amount, error)
 }
+
+// BlockStore defines the interface for database block operations.
+type BlockStore interface {
+	// GetBlock retrieves a block by its height from the database.
+	GetBlock(ctx context.Context, height int32) (Block, error)
+
+	// GetBlockTime retrieves just the timestamp of a block by its height.
+	GetBlockTime(ctx context.Context, height int32) (int64, error)
+
+	// CreateBlock creates a new block into the database.
+	CreateBlock(ctx context.Context, block Block) error
+
+	// DeleteBlock removes a block from the database by its height.
+	DeleteBlock(ctx context.Context, height int32) error
```

Comment:
If we want to have this interface, can we rework the methods here to respect the `xxxParams` and `xxxQueries` pattern? I think we can consolidate these methods, and we are missing a "list blocks" so the caller can decide which methods to use.